vou/person.py

```diff
@@ -220,25 +227,117 @@ def compute_OD_dose_reduction(self):
         else:
             return dose_reduction
 
+   
+
     def will_increase_dose(
-        self, effect_window: int = 20, increase_threshold: float = 0.4,
+        self, t:int, effect_window: int = 20, increase_threshold: float = 0.4,
     ):
         """
         Checks whether the person will increase their dose. Based on a comparison of
         the average of past dose effects to the person's desired dose. Effect window
         and increase threshold are calibrated parameters and not intended to be varied
         during simulation.
+
+        Logic for the source of the increased dose is added here.  If a person will increase
+        their dose for the first time, they will attempt through their primary doctor.  If
+        this is their second time, they will have a 50% chance of trying through a secondary
+        doctor or a dealer.
         """
+        
+        if len(list(self.dose_increase_record.values())) == 0:
+            last_attempt_src = "primary_doctor"
+            last_attempt_success = None
+
+        else:
+            last_attempt_src = list(self.dose_increase_record.values())[-1]['source']
+            last_attempt_success = list(self.dose_increase_record.values())[-1]['success']
+
         if self.dose < 2_000:
+            
             last_n_dose_effects = [
                 self.effect_record[d]
                 for d in self.took_dose[-effect_window:]
                 if d > self.last_dose_increase
             ]
             if len(last_n_dose_effects) > 0:
+                
                 if np.mean(last_n_dose_effects) < (self.dose * increase_threshold):
+                    
                     if self.rng.random() > self.downward_pressure:
-                        return True
+                        
+                        will_increase_dose = True
+                    else:
```

Comment:
One way to avoid these nasty nested conditionals would be to reverse the logic of the conditionals and turn this into a single if, elif, else statement. E.g. (simplified for brevity):
```
if self.dose > 2_000:
    will_increase_dose = False
elif len(last_n_dose_effects) == 0:
    will_increase_dose = False
...
else:
    will_increase_dose = True
```